### DISCUSSION TITLE
Assigning material properties to newly created materials

### DISCUSSION THREAD

**Discussion by Somayaji1995** [ansys/pyaedt#4851] [Q&A] [answered]
I am creating new material using pyAEDT in Icepak with command add_material and specifying orthotropic thermal conductivity, which is working fine , Command I am using is 

New_mat=ipk.materials.add_material("NEW_MAT").thermal_conductivity=['21','21','31'] 

I want to add density as well as specific heat for the same, which i am unable to, can you suggest options for it. 

These are the few options i tried for it which did not work 

New_mat=ipk.materials.add_material("NEW_MAT").thermal_conductivity=['21','21','31'].New_mat.specific_heat("1350")
&
New_mat.specific_heat("1350")

**Comment by gmalinve** [comment]
Hi @Somayaji1995,

Have you tried:
`new_mat = ipk.materials.add_material("NEW_MAT")`
First you initialize the object material _new_mat_  with the _add_material()_ method and then you can access all its properties such as thermal conductivity, specific heat etc.
From your IDE of choice please try to type: `dir(new_mat)`, this will returns all the available properties and methods for this object. 
Then you can access your properties as:
`thermal_conductivity = new_mat.thermal_conductivity=['21','21','31']`
`specific_heat = new_mat.specific_heat("1350")`

Hope this helps,

Regards,

Giulia

**Comment by ansys-satyajeet** [comment, marked as answer]
Hi,

If you must assign properties for density and specific heat, please see the code lines below. It is a good practice to check if the material already exists before creating a new one. The properties of the object `mat` can be obtained using `dir(mat)` as suggested by @gmalinve above.

```python
if not ipk.materials.checkifmaterialexists("matname"):
    mat = ipk.materials.add_material("matname")
    mat.thermal_conductivity = "400"
    mat.mass_density = "8000"
    mat.specific_heat = "700"
```
